From 4d67ddea91d73da9f09ba52cf097cb5a80757860 Mon Sep 17 00:00:00 2001
From: Adam Fontenot <adam.m.fontenot@gmail.com>
Date: Wed, 19 Nov 2014 10:13:17 +0000
Subject: [PATCH 2/2] Fixed problems causing tray icon not to appear in KDE:
 Typo in filename (wl00.png -> wl100.png) Check tray icon geometry values are
 >= 0, not >0.

---
 apps/cmstapp/code/control_box/controlbox.cpp | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/apps/cmstapp/code/control_box/controlbox.cpp b/apps/cmstapp/code/control_box/controlbox.cpp
index ab0c86c..ab70570 100755
--- a/apps/cmstapp/code/control_box/controlbox.cpp
+++ b/apps/cmstapp/code/control_box/controlbox.cpp
@@ -952,16 +952,16 @@ void ControlBox::toggleTrayIcon(bool b_checked)
     } // if
     else {
       ui.pushButton_minimize->setDisabled(false);
-			// QT5.3 and XFCE don't play nicely.  Hammer the XFCE tray up to
-			// maxtries to get a valid icon geometry 
-			const int maxtries = 125;
-			for (int i = 0; i < maxtries; ++i) {
-				trayicon->setVisible(true);
-				if (trayicon->geometry().left() > 0 && trayicon->geometry().top() > 0) return;
-				trayicon->setVisible(false);
-			}	// hammer loop
-			qDebug() << QString("Failed to get a valid icon from the systemtray in %1 tries").arg(maxtries);
-			ui.pushButton_minimize->setDisabled(true);
+            // QT5.3 and XFCE don't play nicely.  Hammer the XFCE tray up to
+            // maxtries to get a valid icon geometry 
+            const int maxtries = 125;
+            for (int i = 0; i < maxtries; ++i) {
+                trayicon->setVisible(true);
+                if (trayicon->geometry().left() >= 0 && trayicon->geometry().top() >= 0) return;
+                trayicon->setVisible(false);
+            }   // hammer loop
+            qDebug() << QString("Failed to get a valid icon from the systemtray in %1 tries").arg(maxtries);
+            ui.pushButton_minimize->setDisabled(true);
     } // else
   } //if
   
@@ -1519,7 +1519,7 @@ void ControlBox::assembleTrayIcon()
           quint8 str = services_list.at(0).objmap.value("Strength").value<quint8>();
           if (b_useicontheme) trayicon->setIcon(QIcon::fromTheme("network-transmit-receive", QIcon(":/icons/images/systemtray/wl00.png")) );
           else {
-          if (str > 80 ) trayicon->setIcon(QIcon(":/icons/images/systemtray/wl00.png"));
+          if (str > 80 ) trayicon->setIcon(QIcon(":/icons/images/systemtray/wl100.png"));
           else if (str > 60 )  trayicon->setIcon(QIcon(":/icons/images/systemtray/wl075.png"));
             else if (str > 40 )  trayicon->setIcon(QIcon(":/icons/images/systemtray/wl050.png"));
               else if (str > 20 )  trayicon->setIcon(QIcon(":/icons/images/systemtray/wl025.png"));
-- 
2.1.3

